include/ghex/transport_layer/ucx/communicator.hpp

```diff
@@ -160,25 +155,22 @@ namespace gridtools {
                         gridtools::ghex::tl::cb::progress_status status;
                         int p = 0;
                         p+= ucp_worker_progress(m_ucp_sw);
-                        p+= ucp_worker_progress(m_ucp_sw);
-                        p+= ucp_worker_progress(m_ucp_sw);
+
```

Comment:
because it gave better performance - @angainor did do some more benchmarks and this now in conjunction with the sched_yield below should be ok, am I correct on this?

Reply:
The multi-progress version has some performance benefits on our Intel-based platform for up to ~8 threads. Now that I tested this on up to 64 threads per  rank I can see that 1) we have to have a `sched_yield` call and 2) we should not call UCX progress more than once. Otherwise I observed lots of performance 'lock-ups', probably because one (or few) threads would dominate the execution and not let others progress the communication.